addons/web/static/src/js/fields/signature.js

```diff
@@ -230,19 +231,13 @@ var FieldBinarySignature = AbstractFieldBinary.extend({
                 noInputName: true,
             };
 
-            if (this.nodeOptions.full_name) {
-                var $search = $("[name='" + this.nodeOptions.full_name + "']");
-                // If full_name correspond to field in read-only mode
-                var signName = $($search).text();
-
-                // If full_name correspond to field to fill
-                if (!signName && $($search)[0]) {
-                    signName = $($search)[0].value;
-                }
-                if (!signName) {
-                    // If full_name is in the recordData. Last option because this value might not be the last value.
-                    signName = this.recordData[this.nodeOptions.full_name] || '';
-                }
+            if (this.nodeOptions.full_name && this.fields[this.nodeOptions.full_name]) {
+                var signName;
+                if (this.fields[this.nodeOptions.full_name].type === 'many2one' && this.recordData[this.nodeOptions.full_name]) {
```

Comment:
Same here, I would simplify to:

`if (this.fields[this.nodeOptions.full_name].type === 'many2one')`

Reply:
When m2o field provided in 'full_name' is not there in the view, it won't be available in recordData and throw a traceback if we remove this extra check here. If we want to simplify the condition, may be can utilize `fields_in_view` [here](https://github.com/odoo/enterprise/pull/4245/files#diff-71e54d51fbb30b58665ff757a89105e3R160) instead of `orderered_fields`. Your call :)

Reply:
Yep, I think it's a better idea to use ´fields_in_view` indeed. Good call.

Reply:
Ideally, we should use `fieldDependencies` so the field doesn't need to be in the view but this attribute doesn't work with a dynamic attribute I think :/